### PR TITLE
New addon setup

### PR DESCRIPTION
**Pull request by sneridagh**

### CODE REVIEW

**Comment by sneridagh** [comment]
@ericof I moved the add-on to the cookieplone template, and CI is yelling at me at coverage and circular... why?

File: setup.py [hunk 1]
```diff
@@ -1,66 +1,80 @@
...
+        "Programming Language :: Python :: 3.11",
         "Operating System :: OS Independent",
         "License :: OSI Approved :: GNU General Public License v2 (GPLv2)",
     ],
-    keywords="Python Plone",
-    author="kitconcept GmbH",
-    author_email="info@kitconcept.com",
-    url="https://pypi.python.org/pypi/kitconcept.seo",
+    keywords="Python Plone CMS",
+    author="Plone Community",
```
Comment (davisagli) [inline setup.py:36]:
```suggestion
    author="kitconcept GmbH",
```

File: setup.py [hunk 2]
```diff
@@ -1,66 +1,80 @@
...
+    author_email="collective@plone.org",
```
Comment (davisagli) [inline setup.py:37]:
```suggestion
    author_email="info@kitconcept.com",
```

File: setup.py [hunk 3]
```diff
@@ -1,66 +1,80 @@
...
+    url="https://github.com/collective/kitconcept.seo",
```
Comment (davisagli) [inline setup.py:38]:
```suggestion
    url="https://github.com/kitconcept/kitconcept.seo",
```

File: setup.py [hunk 4]
```diff
@@ -1,66 +1,80 @@
...
+    project_urls={
+        "PyPI": "https://pypi.python.org/pypi/kitconcept.seo",
+        "Source": "https://github.com/collective/kitconcept.seo",
+        "Tracker": "https://github.com/collective/kitconcept.seo/issues",
```
Comment (davisagli) [inline setup.py:42]:
```suggestion
        "Source": "https://github.com/kitconcept/kitconcept.seo",
        "Tracker": "https://github.com/kitconcept/kitconcept.seo/issues",
```

File: src/kitconcept/seo/controlpanel/configure.zcml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<configure
+    xmlns="http://namespaces.zope.org/zope"
+    xmlns:browser="http://namespaces.zope.org/browser"
+    xmlns:plone="http://namespaces.plone.org/plone"
+    i18n_domain="kitconcept.seo"
+    >
+
+</configure>
```
Comment (davisagli) [inline src/kitconcept/seo/controlpanel/configure.zcml:8]: Why are you adding empty zcml files?

File: src/kitconcept/seo/events.py
Comment (davisagli) [inline src/kitconcept/seo/events.py:1]: This is supposed to be part of #17

File: src/kitconcept/seo/serializers/summary.py [hunk 6]
```diff
@@ -0,0 +1,10 @@
+from plone.restapi.interfaces import IJSONSummarySerializerMetadata
+from zope.interface import implementer
+
+
+@implementer(IJSONSummarySerializerMetadata)
+class JSONSummarySerializerMetadata:
+    """Additional metadata to be exposed on listings."""
+
+    def default_metadata_fields(self):
+        return {"image_field", "image_scales", "effective", "Subject"}
```
Comment (davisagli) [inline src/kitconcept/seo/serializers/summary.py:10]: @sneridagh Why does this belong in kitconcept.seo? These should be added in plone.volto (the first 2 already are) if we need them for listings.

File: src/kitconcept/seo/testing.py [hunk 7]
```diff
@@ -1,44 +1,38 @@
...
+class Layer(PloneSandboxLayer):
     defaultBases = (PLONE_APP_CONTENTTYPES_FIXTURE,)
 
     def setUpZope(self, app, configurationContext):
         # Load any other ZCML that is required for your tests.
         # The z3c.autoinclude feature is disabled in the Plone fixture base
         # layer.
+        import plone.restapi
+
+        self.loadZCML(package=plone.restapi)
```
Comment (davisagli) [inline src/kitconcept/seo/testing.py:20]: Is this really needed? It's already in dependencies.zcml

File: src/kitconcept/seo/tests/test_behaviors_seo.py
Comment (davisagli) [inline src/kitconcept/seo/tests/test_behaviors_seo.py:1]: Did you intend to remove these tests?

**Review by davisagli — CHANGES_REQUESTED** [review]

**Comment by davisagli** [comment]
@sneridagh I fixed the coverage check and disabled the circular check (it's complaining about cycles in Plone core) but there are some other things that aren't ready yet.

File: README.md [hunk 8]
```diff
@@ -0,0 +1,36 @@
+# kitconcept.seo
+
+A new addon for Plone
```
[the remark this answers is not in the document]
Comment (sneridagh) [reply README.md]: Done, sorry for the missing pieces.

File: setup.py [hunk 9]
```diff
@@ -1,66 +1,80 @@
...
+{Path("CONTRIBUTORS.md").read_text()}\n
+{Path("CHANGES.md").read_text()}\n
+"""
 
 
 setup(
     name="kitconcept.seo",
-    version="2.0.2.dev0",
-    description="SEO optimizations plugin for Plone",
+    version="1.0.0a0",
```
[the remark this answers is not in the document]
Comment (sneridagh) [reply setup.py]: Fixed

File: setup.py [hunk 10]
```diff
@@ -1,66 +1,80 @@
...
+    description="A new addon for Plone.",
```
[the remark this answers is not in the document]
Comment (sneridagh) [reply setup.py]: Fixed

File: setup.py [hunk 11]
```diff
@@ -1,66 +1,80 @@
...
     long_description=long_description,
-    # Get more from https://pypi.python.org/pypi?%3Aaction=list_classifiers
+    long_description_content_type="text/markdown",
     classifiers=[
-        "Development Status :: 5 - Production/Stable",
+        "Development Status :: 3 - Alpha",
```
[the remark this answers is not in the document]
Comment (sneridagh) [reply setup.py]: Fixed

File: setup.py [hunk 12]
```diff
@@ -1,66 +1,80 @@
...
         "Environment :: Web Environment",
         "Framework :: Plone",
-        "Framework :: Plone :: 5.2",
```
[the remark this answers is not in the document]
Comment (sneridagh) [reply setup.py:27]: @davisagli we need to talk about it, I think we should. However, it's not that we are not breaking anything, is that we won't support it anymore.

File: src/kitconcept/seo/controlpanel/configure.zcml (hunk 5 above)
Comment (sneridagh) [reply src/kitconcept/seo/controlpanel/configure.zcml]: cleaned up

File: setup.py (hunk 1 above)
Comment (sneridagh) [reply setup.py]: Fixed

File: setup.py (hunk 2 above)
Comment (sneridagh) [reply setup.py]: Fixed

File: setup.py (hunk 3 above)
Comment (sneridagh) [reply setup.py]: Fixed

File: setup.py (hunk 4 above)
Comment (sneridagh) [reply setup.py]: Fixed

File: src/kitconcept/seo/events.py
Comment (sneridagh) [reply src/kitconcept/seo/events.py:1]: cleaned up

File: src/kitconcept/seo/serializers/summary.py (hunk 6 above)
Comment (sneridagh) [reply src/kitconcept/seo/serializers/summary.py:10]: Uh, this comes from cookieplone templates: https://github.com/plone/cookieplone-templates/blob/main/backend_addon/%7B%7B%20cookiecutter.__folder_name%20%7D%7D/src/%7B%7B%20cookiecutter.__package_namespace%20%7D%7D/%7B%7B%20cookiecutter.__package_name%20%7D%7D/serializers/summary.py

@ericof why is that in there? Should we remove it?

Comment (sneridagh) [reply src/kitconcept/seo/serializers/summary.py:10]: @davisagli removing it for now

File: src/kitconcept/seo/testing.py (hunk 7 above)
Comment (sneridagh) [reply src/kitconcept/seo/testing.py:20]: Fixed

File: src/kitconcept/seo/tests/test_behaviors_seo.py
Comment (sneridagh) [reply src/kitconcept/seo/tests/test_behaviors_seo.py:1]: Yes.

**Comment by sneridagh** [comment]
@davisagli sorry for the crappy PR, I stopped working on it when I saw all the issues in CI, now should be ready for review!

Regarding the support for 5.2, no clue if we can do that with mxdev in an easy way, so I would tend to remove support.

Regarding the tests, I'm still not comfortable with pytest tests, maybe it's time for you and @ericof to make a small training session to all the team. Also, the existing couple of tests were quite basic.

Let's talk this evening.

**Comment by tisto** [comment]
@sneridagh @davisagli if supporting Plone 5.2 requires us to put effort into it, let's drop the support and do a major version bumb. If we need Plone 5.2 we can still use the old version and even make new releases in that branch if necessary.

**Comment by tisto** [comment]
@sneridagh we need to add two test dependencies to make the CI pass:

```
Requirements that should be test requirements
=============================================
     plone.app.contenttypes
     plone.restapi
```

File: src/kitconcept/seo/tests/test_behaviors_seo.py
Comment (davisagli) [reply src/kitconcept/seo/tests/test_behaviors_seo.py:1]: @sneridagh I'll try to write a new test using pytest and plone.restapi. We need at least a basic test to show that the behavior actually adds the expected fields.

**Review by davisagli — COMMENTED** [review]
Looking a lot better now, but I don't want to lose the tests.

**Comment by davisagli** [comment]
@tisto @sneridagh Actually we already stopped supporting Plone 5 in version 2.0.0, according to the changelog.

**Review by davisagli — APPROVED** [review]

**Pull request merged**